### PR TITLE
Fix #250: Improve isolation of enrollment and onboarding server

### LINKED ISSUE #250: Improve isolation of enrollment and oboarding server
Improve isolation of enrollment and oboarding server. E.g. at Tomcat, for enrollment server, OnboardingServerApplication is initialized via `ServletInitializer`.

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: enrollment-server/src/main/resources/application.properties [hunk 1]
```diff
@@ -75,6 +75,7 @@ powerauth.service.security.clientSecret=
 # Enrollment Server Configuration
 enrollment-server.mtoken.enabled=true
 enrollment-server.activation-spawn.enabled=false
+#enrollment-server.onboarding.enabled=true
```
Comment (banterCZ) [inline enrollment-server/src/main/resources/application.properties:78]: Where should I document this?

Comment (romanstrobl) [reply enrollment-server/src/main/resources/application.properties:78]: In `/docs`, thank you.

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/onboarding/activation/OnboardingComponentsConfiguration.java [hunk 2]
```diff
@@ -17,35 +17,45 @@
...
-@Configuration
-// TODO (racansky, 2022-06-21) improve isolation, enrollment includes all classes from onboarding
+@Slf4j
 public class OnboardingComponentsConfiguration {
 
+    /**
+     * No-arg constructor.
+     */
+    public OnboardingComponentsConfiguration() {
+        log.info("Onboarding feature turned on, configuring enrollment components specific for onboarding.");
```
Comment (romanstrobl) [inline enrollment-server/src/main/java/com/wultra/app/enrollmentserver/onboarding/activation/OnboardingComponentsConfiguration.java:56]: We usually use `logger`.

File: enrollment-server-onboarding/pom.xml [hunk 3]
```diff
@@ -160,9 +156,6 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-war-plugin</artifactId>
-                <configuration>
-                    <attachClasses>true</attachClasses>
```
Comment (romanstrobl) [inline enrollment-server-onboarding/pom.xml:164]: Hurray!

File: enrollment-server/pom.xml [hunk 4]
```diff
@@ -52,14 +52,6 @@
             <artifactId>enrollment-server-common</artifactId>
         </dependency>
 
-        <!-- TODO (racansky, 2022-06-21) not sufficient isolation, enrollment shares some logic from onboarding but including all classes -->
-        <dependency>
-            <groupId>com.wultra.security</groupId>
-            <artifactId>enrollment-server-onboarding</artifactId>
-            <version>${project.version}</version>
-            <classifier>classes</classifier>
-        </dependency>
```
Comment (romanstrobl) [inline enrollment-server/pom.xml:61]: 👏

**Review by romanstrobl — APPROVED** [review]
No big surprises for me, let's deploy the fix and keep running with it for a while.

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/onboarding/activation/OnboardingComponentsConfiguration.java (hunk 2 above)
Comment (banterCZ) [reply enrollment-server/src/main/java/com/wultra/app/enrollmentserver/onboarding/activation/OnboardingComponentsConfiguration.java:56]: I will introduce lombok way later.

**Pull request merged**